File: app/src/main/java/jose/com/bookworm/view/BookViewHolder.java
package jose.com.bookworm.view;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import jose.com.bookworm.R;
import jose.com.bookworm.model.Book;

/**
 * Created by Joe on 10/30/17.
 */

public class BookViewHolder extends RecyclerView.ViewHolder {

    private TextView textView;

    public BookViewHolder(View itemView) {
        super(itemView);
        textView = itemView.findViewById(R.id.title_text_view);
    }

    public void bind(Book book) {
        textView.setText(book.getTitle());
    }
}

File: app/src/main/java/jose/com/bookworm/viewmodel/BookViewHolder.kt
package jose.com.bookworm.viewmodel

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView

import jose.com.bookworm.R
import jose.com.bookworm.model.Book

/**
 * Created by Joe on 10/30/17.
 */

class BookViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

  private val textView: TextView

  init {
    textView = itemView.findViewById(R.id.title_text_view)
  }

  fun bind(book: Book) {
    textView.text = book.title
  }

  fun bind(string: String) {
    textView.text = string
  }
}

